### PR TITLE
Added ability to choose road types in editor RMG options

### PR DESCRIPTION
**Pull request by wb180**
Screenshot_20240916_173820

### CODE REVIEW

File: mapeditor/jsonutils.cpp [hunk 1]
```diff
@@ -108,10 +108,11 @@ JsonNode toJson(QVariant object)
 
 	if(object.canConvert<QVariantMap>())
 		ret.Struct() = VariantToMap(object.toMap());
-	else if(object.canConvert<QVariantList>())
-		ret.Vector() = VariantToList(object.toList());
 	else if(object.userType() == QMetaType::QString)
 		ret.String() = object.toString().toUtf8().data();
+	//QStirng can be converted to QVariantList, need to check first
+	else if(object.canConvert<QVariantList>())
+		ret.Vector() = VariantToList(object.toList());
```
Comment (IvanSavenko) [inline mapeditor/jsonutils.cpp:115]: I remember there was some problem with Qt5/Qt6 having different behavior. 
Are you testing with Qt5 or Qt6?

Comment (kambala-decapitator) [reply mapeditor/jsonutils.cpp:115]: you probably mean deprecated toSet / toList methods on QSet / QList, but here it's QVariant

Comment (IvanSavenko) [reply mapeditor/jsonutils.cpp:115]: No, I mean this commit:
https://github.com/vcmi/vcmi/commit/01af231084d5ed23dfad5571eb633624d56f0cf2

Yay for more copy-pasted code in editor, which now diverges even more for no reason...

In any case, I think it effectively the same change, albeit done a little bit differently - starting from Qt6, QVariant that contains QString can be converted into QList<char>.

I see 3 options:
- keep change from this PR as it
- copy-paste entire launcher/jsonutils into editor/jsonutils to bring them back in sync
- remove copy-pasted files. Add some sort of "vcmiqt" static library?

Comment (wb180) [reply mapeditor/jsonutils.cpp]: > No, I mean this commit: [01af231](https://github.com/vcmi/vcmi/commit/01af231084d5ed23dfad5571eb633624d56f0cf2)
> 
> Yay for more copy-pasted code in editor, which now diverges even more for no reason...
> 
> In any case, I think this is effectively the same change, albeit done a little bit differently - starting from Qt6, QVariant that contains QString can be converted into QList of chars
> 
> I see 3 options:
> 
>     * keep change from this PR as it
> 
>     * copy-paste entire launcher/jsonutils into editor/jsonutils to bring them back in sync

Updated editor/jsonutils with launcher/jsonutils

> 
>     * remove copy-pasted files. Add some sort of "vcmiqt" static library?

Think that requires separate PR.

**Pull request merged**